### PR TITLE
[REG-1831] Expand user configuration capabilities of action manager

### PR DESCRIPTION
**Pull request by svolokh**
## Overview
This gives the user more fine-grained configuration capabilities over the actions in the action manager. Previously, the user could only enable/disable actions in the action manager UI. With this PR, users can now modify individual properties of the actions, such as their parameter range or mouse movement magnitude. This enables more flexibility in defining the monkey bot's behavior, such as only moving right in platformer games.

## Key Changes
1. Introduced the `RGActionProperty` attribute and "ActionProperties" option in the action manager settings JSON. This enables serializing individual property changes to the actions in the settings and applying these afterwards on top of the base analysis results.
2. Updated the action manager UI to iterate over all action properties annotated with `RGActionProperty` and create UI controls for the those that are marked a user configurable.

## Loom
https://www.loom.com/share/607d241e4f1044f9affb67ddb0ddfe47

---

Find the pull request instructions [here](https://www.notion.so/regressiongg/Pull-Request-Process-0d57a7eb582a446983edfacafa406f1e?pvs=4)

Every reviewer and the owner of the PR should consider these points in their request (feel free to copy this checklist so you can fill it out yourself in the overall PR comment)

- [ ] The code is extensible and backward compatible
- [ ] New public interfaces are extensible and open to backward compatibility in the future
- [ ] If preparing to remove a field in the future (i.e. this PR removes an argument), the argument stays but is no longer functional, and attaches a deprecation warning. A linear task is also created to track this deletion task.
- [ ] Non-critical or potentially modifiable arguments are optional
- [ ] Breaking changes and the approach to handling them have been verified with the team (in the Linear task, design doc, or PR itself)
- [ ] The code is easy to read
- [ ] Unit tests are added for expected and edge cases
- [ ] Integration tests are added for expected and edge cases
- [ ] Functions and classes are documented
- [ ] Migrations for both up and down operations are completed
- [ ] A documentation PR is created and being reviewed for anything in this PR that requires knowledge to use
- [ ] Implications on other dependent code (i.e. sample games and sample bots) is considered, mentioned, and properly handled
- [ ] Style changes and other non-blocking changes are marked as non-blocking from reviewers

### CODE REVIEW

File: src/gg.regression.unity.bots/Runtime/Scripts/ActionManager/README.md [hunk 1]
```diff
@@ -55,13 +55,10 @@ public class RandomBot : MonoBehaviour, IRGBot
         // Start an action manager session (must be called before using any of the action manager API)
         RGActionManager.StartSession(this);
         
-        // Iterate over all the actions identified by the static analysis
+        // Iterate over all the actions
```
Comment (RG-nAmKcAz) [inline src/gg.regression.unity.bots/Runtime/Scripts/ActionManager/README.md:58]:
```suggestion
        // Iterate over all the enabled actions
```

**Review by RG-nAmKcAz — APPROVED** [review]

File: src/gg.regression.unity.bots/Editor/Scripts/ActionManager/RGActionManagerUI.cs [hunk 2]
```diff
@@ -237,22 +236,613 @@ private void UpdateGUI()
...
+            }
+        }
+
+        private VisualElement CreateControlForProperty(RGActionPropertyInstance prop, object origValue, Action valueChangedCallback, out Action onPropReset)
+        {
+            const int numericFieldWidth = 60;
+            var valueType = prop.GetValue().GetType();
+            if (valueType == typeof(string))
+            {
+                TextField textField = new TextField();
```
Comment (vontell) [inline src/gg.regression.unity.bots/Editor/Scripts/ActionManager/RGActionManagerUI.cs:274]: All of this UI code reminds me of the days before React and Angular on the web lol

**Review by vontell — APPROVED** [review]
I'm not sure how I missed this, but where exactly in the code is the part that constrains the action being taken based on the action property settings? No changes though!

**Comment by svolokh** [comment]
@vontell This is the key line of code that filters the actions based on the settings:
https://github.com/Regression-Games/RGUnityBots/blob/54ef7d307795a36bbdcc766983e1b507258ec62b/src/gg.regression.unity.bots/Runtime/Scripts/ActionManager/RGActionManager.cs#L251

The `ApplySettings()` method filters and applies property changes to the original actions determined from the analysis.

File: src/gg.regression.unity.bots/Editor/Scripts/ActionManager/RGActionManagerUI.cs (hunk 2 above)
Comment (svolokh) [reply src/gg.regression.unity.bots/Editor/Scripts/ActionManager/RGActionManagerUI.cs:274]: Unity does have the UXML layout language, but pretty much everything in the action manager UI is dynamically constructed from the analysis outcome, so just constructing the components in code was easier.

**Pull request merged**